python/ray/autoscaler/v2/tests/test_sdk.py

```diff
@@ -460,6 +460,138 @@ def verify_cluster_no_node():
     wait_for_condition(verify_cluster_no_node)
 
 
+def test_idle_node_blocked(ray_start_cluster):
+    cluster = ray_start_cluster
+    cluster.add_node(num_cpus=1)
+    ray.init(address=cluster.address)
+    cluster.add_node(num_cpus=1)
+
+    stub = _autoscaler_state_service_stub()
+
+    # We don't have node id from `add_node` unfortunately.
+    def nodes_up():
+        nodes = list_nodes()
+        assert len(nodes) == 2
+        return True
+
+    wait_for_condition(nodes_up)
+
+    head_node_id, worker_node_ids = get_node_ids()
+    node_id = worker_node_ids[0]
+
+    def verify_cluster_idle():
+        state = get_cluster_resource_state(stub)
+        assert_node_states(
+            state,
+            [
+                ExpectedNodeState(
+                    head_node_id, NodeStatus.IDLE, lambda idle_ms: idle_ms > 0
+                ),
+                ExpectedNodeState(
+                    node_id, NodeStatus.IDLE, lambda idle_ms: idle_ms > 0
+                ),
+            ],
+        )
+        return True
+
+    async_wait_for_condition(verify_cluster_idle)
```

Comment:
i think all the `async_wait_for_condition` are just returning immediately w/o awaiting on the awaitables?

Reply:
oof